source/_posts/angular-csrf-protection-implementation.md

```diff
@@ -149,44 +149,47 @@ The secure flag prevents this attack by the browser guaranteeing that the cookie
 
 Finally, the last security flag that should be set in the cookie is a maxAge. This ensures that the cookie is renewed with a new token so that the attacker is less likely to gain access to the cookie in other methods; like having physical access to the computer.
 
+##### CSURF Package Vulnerability
+
+For those who have implimented a CSRF prevention method using CSURF, a popular Node.js CSRF protection middleware, there has been a rediscovered vulnerability. The `cookie: true` flag set is the trouble, and in short, an attacker can use cookie tossing (setting a cookie from a subdomain) to use a valid (and signed) pair of cookies (_csrf + token) to bypass the anti-csrf mechanism. This vulnerability is not present when `cookie: false` is set; however in abundance of caution the package author has depricated this module. 
+
+Luckly for us, Angular has a built in mechanism to handle CSRF automatically. In the case with Angular, you can set a `XSRF-TOKEN`cookie to allow for protection to take place.  
+
 The following is an example of the implementation of these methods in NodeJS. The user makes a GET request to the '/form' API endpoint and then when the user fills out the form, using angular form control and form builder, submits the data to the server via a POST request at the '/process' API endpoint.
 
 ``` typescript
 var cookieParser = require('cookie-parser')
-var csrf = require('csurf')
+// var csrf = require('csurf') <-- DO NOT USE (See above 'CSURF Package Vulnerability')
 var bodyParser = require('body-parser')
 var express = require('express')
 
-// setup route middlewares
-var csrfProtection = csrf({ 
- cookie: true,
- secure: true,
- maxAge: 3600,
- httpOnly: true,
- sameSite: strict
-});
+
  var parseForm = bodyParser.urlencoded({ extended: false })
  
  // create express app
  var app = express()
  
  // parse cookies
  app.use(cookieParser())
+
+ app.all('*', function(res, req){
+    res.cookie('XSRF-TOKEN', req.csrfToken()) // <-- remember to change the cookie name to match what was used in the HttpClientXsrfModule settings
+    res.render('index')
+ })
  
- app.get('/form', csrfProtection, function (req, res) {
- // pass the csrfToken to the view
- res.render('send', { csrfToken: req.csrfToken() })
+ app.get('/form', function (req, res) {
+    res.render('send')
```

Comment:
I think we can remove that `/form` handler if we use `app.all('*')` to pass the token.